cmake/mode.common.cmake

```diff
@@ -24,9 +24,8 @@ add_compile_options(
   ${_supported_warnings})
 
 function(default_target_arch arch)
-  set(x86_instruction_sets i386 i686 x86_64)
-  if(CMAKE_SYSTEM_PROCESSOR IN_LIST x86_instruction_sets)
-    set(${arch} "westmere" PARENT_SCOPE)
+  if(CMAKE_SYSTEM_PROCESSOR STREQUAL "x86_64")
+    set(${arch} "broadwell" PARENT_SCOPE)
```

Comment:
Let's try x86-64-v3 if cmake/clang support it.

Reply:
This requires adding '-mpclmul' as x86-64-v3 does not contain it:
```
clang++ -MD -MT build/release/db/commitlog/commitlog.o -MF build/release/db/commitlog/commitlog.o.d -std=c++23 -I/home/ykaul/github/scylladb/seastar/include -I/home/ykaul/github/scylladb/build/release/seastar/gen/include -Werror=unused-result -DSEASTAR_API_LEVEL=7 -DSEASTAR_SSTRING -DSEASTAR_LOGGER_COMPILE_TIME_FMT -DSEASTAR_SCHEDULING_GROUPS_COUNT=16 -DSEASTAR_LOGGER_TYPE_STDOUT -DFMT_SHARED -I/usr/include/p11-kit-1 -DWITH_GZFILEOP -ffile-prefix-map=/home/ykaul/github/scylladb=. -march=x86-64-v3 -ffunction-sections -fdata-sections  -O3 -mllvm -inline-threshold=2500 -fno-slp-vectorize -DSCYLLA_BUILD_MODE=release -g -gz -Xclang -fexperimental-assignment-tracking=disabled -iquote. -iquote build/release/gen -std=gnu++23  -ffile-prefix-map=/home/ykaul/github/scylladb=. -march=x86-64-v3 -DBOOST_ALL_DYN_LINK    -fvisibility=hidden -isystem abseil -Wall -Werror -Wextra -Wimplicit-fallthrough -Wno-mismatched-tags -Wno-c++11-narrowing -Wno-overloaded-virtual -Wno-unused-parameter -Wno-unsupported-friend -Wno-missing-field-initializers -Wno-deprecated-copy -Wno-psabi -Wno-enum-constexpr-conversion -Wno-error=deprecated-declarations -DXXH_PRIVATE_API -DSEASTAR_TESTING_MAIN  -c -o build/release/db/commitlog/commitlog.o db/commitlog/commitlog.cc
In file included from db/commitlog/commitlog.cc:51:
In file included from ./utils/crc.hh:45:
./utils/clmul.hh:43:9: error: '__builtin_ia32_pclmulqdq128' needs target feature pclmul
   43 |     p = _mm_clmulepi64_si128(p, p, 0x01);
      |         ^
/usr/local/lib64/clang/18/include/__wmmintrin_pclmul.h:45:13: note: expanded from macro '_mm_clmulepi64_si128'
   45 |   ((__m128i)__builtin_ia32_pclmulqdq128((__v2di)(__m128i)(X), \
      |             ^
1 error generated.
```